### PR TITLE
Adding transaction classification notebooks

### PR DESCRIPTION
**Pull request by colin-jarvis**
I've made two examples of transaction classification using GPT-3, one with multiclass classification and one using clustering on an unlabelled dataset.

I've also included the source dataset used in the multiclass classification notebook, plus a set of labelled examples I made based on it.

I added a .gitignore to the repo as well

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -127,3 +127,9 @@ dmypy.json
 
 # Pyre type checker
 .pyre/
+
+# helpers
+*helpers.py
```
Comment (BorisPower) [inline .gitignore:132]: could you remove this?

File: examples/Multiclass_classification_for_transactions.ipynb [hunk 2]
```diff
@@ -0,0 +1,2201 @@
...
+   "cell_type": "code",
+   "execution_count": 311,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import openai\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "\n",
+    "from helpers import OPENAI_API_KEY\n",
```
Comment (BorisPower) [inline examples/Multiclass_classification_for_transactions.ipynb:44]: Remove this import, and you can replace with:

import os
openai.api_key = os.getenv("OPENAI_API_KEY")

File: examples/Multiclass_classification_for_transactions.ipynb [hunk 3]
```diff
@@ -0,0 +1,2201 @@
...
+    "    prompt = prompt.replace('TRANSACTION_VALUE',str(transaction['Transaction value (£)']))\n",
+    "    \n",
+    "    classification = request_completion(prompt)['choices'][0]['text'].replace('\\n','')\n",
+    "    \n",
+    "    return classification\n",
+    "\n",
+    "import json\n",
+    "\n",
+    "def check_finetune_classes(train_file,valid_file):\n",
+    "\n",
```
Comment (BorisPower) [inline examples/Multiclass_classification_for_transactions.ipynb:227]: Maybe add a very short docstring explaining what this code does

File: examples/Multiclass_classification_for_transactions.ipynb [hunk 4]
```diff
@@ -0,0 +1,2201 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 277,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "zero_shot_prompt = '''You are a data expert working for the National Library of Scotland. \n",
+    "                      You are analysing all transactions over £25,000 in value and classifying them into one of five categories.\n",
```
Comment (BorisPower) [inline examples/Multiclass_classification_for_transactions.ipynb:273]: You have a bunch of whitespace in every line. You should start at the beginning of line (not idented) for every line of the multiline string.

File: examples/Multiclass_classification_for_transactions.ipynb [hunk 5]
```diff
@@ -0,0 +1,2201 @@
...
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from sklearn.ensemble import RandomForestClassifier\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "from sklearn.metrics import classification_report, accuracy_score\n",
+    "\n",
+    "fs_df = pd.read_csv(embedding_path)\n",
```
Comment (BorisPower) [inline examples/Multiclass_classification_for_transactions.ipynb:1091]: you can add a parameter index_col=0 (to get rid of the "Unnamed: 0" column

File: examples/Clustering_for_transaction_classification.ipynb [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+   "source": [
+    "import openai\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "from sklearn.cluster import KMeans\n",
+    "from sklearn.manifold import TSNE\n",
+    "import matplotlib\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "from helpers import OPENAI_API_KEY\n",
```
Comment (BorisPower) [inline examples/Clustering_for_transaction_classification.ipynb:45]: Could you replace this with the same as for the previous notebook?

**Review by BorisPower — CHANGES_REQUESTED** [review]
Could you make those small changes first, then it's good to go! Thanks

File: .gitignore (hunk 1 above)
Comment (colin-jarvis) [reply .gitignore]: This has been removed and the file checked out of Git

**Pull request merged**